homeassistant/components/mqtt/client.py

```diff
@@ -835,8 +835,13 @@ def _exception_message(
         msg: ReceiveMessage,
     ) -> str:
         """Return a string with the exception message."""
+        # if msg_callback is a partial we return the name of the first argument
+        if isinstance(msg_callback, partial):
+            call_back_name = getattr(msg_callback.args[0], "__name__")  # type: ignore[unreachable]
+        else:
+            call_back_name = getattr(msg_callback, "__name__")
```

Comment:
We can use https://github.com/home-assistant/core/blob/569763b7a83411eecfdd1a9f60c4ab7f13bd68b6/homeassistant/core.py#L248 to unwrap the partial

Reply:
But the `func` attribute does not have the `__name__` attribute of the handler.

Reply:
ahh, right. Makes sense.